ingest/validation/validate_metadata.py

```diff
@@ -203,30 +203,48 @@ def validate_cells_unique(metadata):
     return valid
 
 
-def collect_cell_for_ontology(metadatum, row_data, metadata, array=False):
+def collect_cell_for_ontology(metadatum, row, metadata, array=False):
     """Collect ontology info for a single metadatum into CellMetadata.ontology dictionary
     """
     if metadatum.endswith('__unit'):
         ontology_label = metadatum + '_label'
     else:
         ontology_label = metadatum + '__ontology_label'
+    cell_id = row['CellID']
     if array:
+        # Catch unusual edge case where ontology_label column provided but empty
+        # np.isnan does not support string values and generates a TypeError
         try:
-            ontology_dict = dict(zip(row_data[metadatum], row_data[ontology_label]))
-            for id, label in ontology_dict.items():
-                metadata.ontology[metadatum][(id, label)].append(row_data['CellID'])
+            np.isnan(row[ontology_label])
+            for id in row[metadatum]:
+                metadata.ontology[metadatum][(id)].append(cell_id)
         except (TypeError, KeyError):
-            for id in row_data[metadatum]:
-                metadata.ontology[metadatum][(id)].append(row_data['CellID'])
+            # Not all ontology data will have user-provided ontology_label
+            # where provided, collect to provide cross-validation
+            try:
+                ontology_dict = dict(zip(row[metadatum], row[ontology_label]))
+                for id, label in ontology_dict.items():
+                    metadata.ontology[metadatum][(id, label)].append(cell_id)
```

Comment:
Renaming `metadata.ontology[metadatum]` to something shorter would reduce visual noise, which would maintainability for this complex method.

You could append to that more ergonomic variable instead of `metadata.ontology[metadatum]`, then `return` it at the end of the method.